altair/utils/schemapi.py

```diff
@@ -44,6 +44,15 @@
 else:
     from typing_extensions import Self
 
+if TYPE_CHECKING:
+    pass
+
+
+class _PandasTimestamp:
+    def isoformat(self):
+        return "dummy_isoformat"  # Return a dummy ISO format string
```

Comment:
I think this should inherit from a Protocol as a pd.Timestamp is not an instance of `_PandasTimestamp`. You'll then also need to add the `@runtime_checkable` decorator from `typing`. Also, we could directly test for a pandas timestamp in a similar function to `is_pandas_dataframe` to keep these approaches consistent?